catkin_ws/src/cucumber_msgs/include/cucumber_msgs/CucumberContainer.h

```diff
@@ -80,5 +84,23 @@ class CucumberContainer {
 	float getCurvature() {
 		return this->curvature;
 	}
+	
+	/**
+	 * Check if the specified other cucumber is equal to this cucumber.
+	 * Equal, in this case, means that it lies within the bounds set by EQ_DIST, EQ_WEIGHT and EQ_CURVE.
+	 * @param other [CucumberContainer]: the cucumber to check with
+	 * @return True is they are equal, false otherwise
+	 */
+	bool equals(CucumberContainer other) {
+		float dx = this->x - other.x;
+		float dy = this->y - other.y;
+		float dz = this->z - other.z;
+		
+		if (sqrt(dx*dx + dy*dy + dz*dz) > EQ_DIST) return false;
+		if (fabs(this->getWeight() - other.getWeight()) > EQ_WEIGHT) return false;
+		if (fabs(this->getCurvature() - other.getCurvature()) > EQ_CURVE) return false;
```

Comment:
This could be spread over multiple lines, to make it more readable and understandable.